### PR TITLE
[TEST] Test workflow `prepare_phasediff_fmap`

### PR DESCRIPTION
**Pull request by NicolasGensollen**
This PR proposes to add a test for the workflow `prepare_phasediff_fmap`, which is part of the second step of the `DWIPreprocessingUsingPhasediff` pipeline.

It also:

- improves the related documentation
- improves the variable names and connection names
- enables the workflow to write to a custom location

The test is pretty fast and should run in less than a minute.

This requires the data I have generated here : https://github.com/aramis-lab/clinica_data_ci/pull/28

### CODE REVIEW

**Comment by NicolasGensollen** [comment]
The failure is expected since the fast non regression tests are run on PRs and the data for the added test hasn't been merged on the CI machines.

File: clinica/pipelines/dwi_preprocessing_using_fmap/dwi_preprocessing_using_phasediff_fmap_workflows.py [hunk 1]
```diff
@@ -3,25 +3,55 @@
 from nipype.pipeline.engine import Workflow
 
 
-def prepare_phasediff_fmap(name="prepare_phasediff_fmap"):
+def prepare_phasediff_fmap(
+    output_dir: Optional[str] = None,
+    name="prepare_phasediff_fmap",
```
Comment (MatthieuJoulot) [inline clinica/pipelines/dwi_preprocessing_using_fmap/dwi_preprocessing_using_phasediff_fmap_workflows.py:8]:
```suggestion
    name: Optional[str] ="prepare_phasediff_fmap",
```

**Comment by MatthieuJoulot** [comment]
tested with the data and works

**Review by MatthieuJoulot — APPROVED** [review]

File: clinica/pipelines/dwi_preprocessing_using_fmap/dwi_preprocessing_using_phasediff_fmap_workflows.py [hunk 2]
```diff
@@ -3,25 +3,55 @@
 from nipype.pipeline.engine import Workflow
 
 
-def prepare_phasediff_fmap(name="prepare_phasediff_fmap"):
+def prepare_phasediff_fmap(
+    output_dir: Optional[str] = None,
+    name: Optional[str] ="prepare_phasediff_fmap",
```
Comment (NicolasGensollen) [inline clinica/pipelines/dwi_preprocessing_using_fmap/dwi_preprocessing_using_phasediff_fmap_workflows.py:8]:
```suggestion
    name: Optional[str] = "prepare_phasediff_fmap",
```

**Comment by NicolasGensollen** [comment]
Tests are passing with the updated data.
Thanks for the review @MatthieuJoulot !
Merging...

**Pull request merged**